### DISCUSSION TITLE
Nested Array not causing change detection

### DISCUSSION THREAD

**Discussion by Tobjoern** [mobxjs/mobx#3591] [Q&A]
I'm currently struggling to get mobx to react to a nested change in an array.
You can find the repro here: https://github.com/Tobjoern/mobx-reactivity-issue.
In my store, I have:

`export interface ConversationMessage {
  content: string
  id: string
}

export interface Conversation {
  messages: ConversationMessage[]
  user: string
}

conversations: Conversation[] = []
`

And mutate them via:
`
sendMessage(msg: string) {
    this.conversations[0].messages.push({
      content: msg,
      id: Math.random().toString()
    })
  }
`

However, the 'ConversationItem' component does not react to the change at all, it doesn't display the latest message, or even the message count.

Your help is appreciated!

**Comment by mweststrate** [comment]
ConversationItem does not seem marked as observer
